lib/loadEnv.test.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const chai = require('chai');
+const path = require('path');
+const sinon = require('sinon');
+const overrideEnv = require('process-utils/override-env');
+const loadEnv = require('./loadEnv');
+const writeFileSync = require('./utils/fs/writeFileSync');
+const dotenv = require('dotenv');
+const ServerlessError = require('./classes/Error').ServerlessError;
+
+chai.use(require('chai-as-promised'));
+const expect = require('chai').expect;
+
+describe('loadEnv', () => {
+  let restoreEnv;
+
+  before(() => {
+    const stage = 'testing';
+    const stageFileContent = 'FROM_STAGE=valuefromstage';
+    writeFileSync(path.join(process.cwd(), `.env.${stage}`), stageFileContent);
+
+    const defaultFileContent = 'FROM_DEFAULT=valuefromdefault';
+    writeFileSync(path.join(process.cwd(), '.env'), defaultFileContent);
+  });
+
+  beforeEach(() => {
+    restoreEnv = overrideEnv().restoreEnv;
+  });
+
+  afterEach(() => {
+    restoreEnv && restoreEnv();
+  });
+
+  it('should load matching stage env file if present', async () => {
+    await loadEnv('testing');
+    expect(process.env.FROM_DEFAULT).to.be.undefined;
```

Comment:
More logical check is `to.not.have.property()` (theoretically `undefined` can be assigned as value, and that should be not expected)